libs/language-server/src/stdlib/domain/material-science/MatSciValueTypes.jv

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2024 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+constraint DOIFormat on text: value matches /\b10\.\d{4}\/[^\s]+\b/;
+/** DOI Format has been constrained by a standard pattern, eg: 10.1007/xxxx */
```

Comment:
Reference here that the `DOIStandardizer` block allows you to bring dois in a format that matches this.